includes/class-wc-google-gtag-js.php

```diff
@@ -86,27 +86,86 @@ public function variant_data( $data, $product, $variation ) {
 	}
 
 	/**
-	 * Enqueues JavaScript to build the addImpression event
+	 * Returns Javascript string for Google Analytics events
+	 *
+	 * @param string $event The type of event
+	 * @param array  $data  Event data to be sent
+	 * @return string
+	 */
+	public static function get_event_code( string $event, array $data ): string {
+		return sprintf( "%s('event', '%s', %s)", self::tracker_var(), esc_js( $event ), self::format_event_data( $data ) );
+	}
+
+	/**
+	 * Escape and encode event data
+	 *
+	 * @param array $data Event data to processed and formatted
+	 * @return string
+	 */
+	public static function format_event_data( array $data ): string {
+		$data = apply_filters( 'woocommerce_gtag_event_data', $data );
+
+		// Recursively walk through $data array and escape all values that will be used in JS.
+		array_walk_recursive(
+			$data,
+			function( &$value, $key ) {
+				$value = esc_js( $value );
+			}
+		);
+
+		return wp_json_encode( $data );
+	}
+
+	/**
+	 * Returns an array of category names the product is atttributed to
+	 *
+	 * @param  WC_Product $_product  Product to pull info for
+	 * @return array
+	 */
+	public static function product_get_category_line( $_product ) {
+		$out            = [];
```

Comment:
Happy to be using short array syntax, but I do think it's nicer to be consistent. Everywhere else in this PR you use the long syntax.

Reply:
I've reworked the `product_get_category_line` [function](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/2b656ac996cdf886583977a39c77a31d7487db39/includes/class-wc-google-gtag-js.php#L125) instead of having a straight copy from the main analytics class to address the points raised.